plugins/cilium-cni/cmd/cmd.go

```diff
@@ -468,20 +469,22 @@ func (cmd *Cmd) Add(args *skel.CmdArgs) (err error) {
 		}
 	}
 
-	netNs, err := ns.GetNS(args.Netns)
-	if err != nil {
-		return fmt.Errorf("failed to open netns %q: %w", args.Netns, err)
-	}
-	defer netNs.Close()
-
 	res := &cniTypesV1.Result{}
 	configs, err := cmd.cfg.GetConfigurations(ConfigurationParams{log, conf, args, cniArgs})
 	if err != nil {
 		return fmt.Errorf("failed to determine endpoint configuration: %w", err)
 	}
 
+	ns, err := netns.OpenPinned(args.Netns)
+	if err != nil {
+		return fmt.Errorf("opening netns pinned at %s: %w", args.Netns, err)
+	}
+	defer ns.Close()
+
 	for _, epConf := range configs {
-		if err = netns.RemoveIfFromNetNSIfExists(netNs, epConf.IfName()); err != nil {
+		if err = ns.Do(func() error {
+			return link.DeleteByName(epConf.IfName())
```

Comment:
What happens if this link doesn't exist -- will this return an error?

Reply:
`DeleteByName` returns nil if the link doesn't exist, which is the same as the old function.